### PR TITLE
Server: Allow configuring DruidBeamConfig.

### PR DESCRIPTION
**Pull request by gianm**

### CODE REVIEW

**Comment by nishantmonu51** [comment]
:+1:

**Comment by dclim** [comment]
:+1:

File: server/src/main/scala/com/metamx/tranquility/server/config.scala [hunk 1]
```diff
@@ -69,10 +70,41 @@ object config
...
+
+    @Config(Array("druidBeam.firehoseBufferSize"))
+    def firehoseBufferSize: Int = DruidBeamConfig().firehoseBufferSize
+
     @Config(Array("druid.discovery.curator.path"))
     def discoPath: String = "/druid/discovery"
 
     override def discoAnnounce: Option[DiscoAnnounceConfig] = None
+
+    def druidBeamConfig: DruidBeamConfig = DruidBeamConfig.builder()
```
Comment (drcrallen) [inline server/src/main/scala/com/metamx/tranquility/server/config.scala]: Do these need validation steps?

Comment (gianm) [inline server/src/main/scala/com/metamx/tranquility/server/config.scala]: Something crazy will probably happen if you use a negative chunk size or buffer size. If we want to validate that kind of stuff then DruidBeamConfig is probably the right place to do it

**Pull request merged**